tests/integration/test_fim/test_stats_integrity_sync/test_stats_integrity_sync.py

```diff
@@ -0,0 +1,716 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import re
+import time
+from enum import Enum
+from multiprocessing import Process, Manager
+from random import randrange
+from socket import socket, AF_UNIX, SOCK_STREAM
+from struct import pack, unpack
+
+import pytest
+
+from wazuh_testing.tools import WAZUH_PATH, WAZUH_CONF, ALERT_FILE_PATH
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools.monitoring import FileMonitor
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=3)]
+
+# variables
+
+agent_conf = os.path.join(WAZUH_PATH, 'etc', 'shared', 'default', 'agent.conf')
+state_path = os.path.join(WAZUH_PATH, 'var', 'run')
+db_path = '/var/ossec/queue/db/wdb'
+manager_ip = '172.19.0.100'
+setup_environment_time = 1  # Seconds
+state_collector_time = 5  # Seconds
+max_time_for_agent_setup = 180  # Seconds
+max_n_attempts = 20
+tested_daemons = ["wazuh-db", "ossec-analysisd", "ossec-remoted"]
+
+
+class Cases(Enum):
+    """
+        Case 0: In this case, we start from an empty database.
+
+        Case 1: Once the synchronization process is completed,
+        a random checksum in the database will be changed.
+
+        Case 2: Modifies all the checksums of the database entries.
+    """
+    case0 = 0
+    case1 = 1
+    case2 = 2
+
+
+def db_query(agent, query):
+    """Run a query against wazuh_db for a specific agent.
+
+    Parameters
+    ----------
+    agent : str
+        Database identifier.
+    query : str
+        Query to be executed.
+
+    Returns
+    -------
+    str
+        Return the wazuh-db's socket response.
+
+    """
+    sock = socket(AF_UNIX, SOCK_STREAM)
+    sock.connect(db_path)
+
+    msg = f'agent {agent} sql {query}'.encode()
+    sock.send(pack(f"<I{len(msg)}s", len(msg), msg))
+
+    length = unpack("<I", sock.recv(4))[0]
+    return sock.recv(length).decode(errors='ignore')
```

Comment:
Socket must be closed. The send and recv function may not send or receive all bytes, so we need to ensure the message is fully sent/received .